finch/src/modules/base/__tests__/crud-controller.test.js

```diff
@@ -0,0 +1,126 @@
+import express from 'express';
+import httpMocks from 'node-mocks-http';
+
+import CrudController from '../crud-controller.js';
+
+describe('CrudController', () => {
+  describe('#create', () => {
+    test('should call service#create() method with request payload',async () => {
+
+      const samplePayload = {
+        key1: 'value1',
+        key2: 'value2',
+      };
+      const mockReq = httpMocks.createRequest({
+        body: samplePayload,
+      });
+      const mockRes = httpMocks.createResponse();
+
+      const mockService = {
+        create: jest.fn().mockReturnValueOnce(samplePayload),
+      };
+
+      const controller = new CrudController({});
+      controller.service = mockService;
+      await controller.create(mockReq, mockRes);
+
+      expect(mockService.create).toHaveBeenCalledTimes(1);
+      expect(mockService.create).toHaveBeenCalledWith(samplePayload);
+      expect(mockRes.statusCode).toBe(201);
+      expect(mockRes._isJSON()).toBe(true);
+      expect(mockRes._isEndCalled()).toBe(true);
+      expect(mockRes._getJSONData()).toStrictEqual({
+        item: samplePayload,
+        success: true,
+      });
+
+    });
+  });
+
+  describe('#getById', () => {
+    test('should call service#getById() method with id in request.params',async () => {
+      const mockReq = httpMocks.createRequest({
+        params: { id: 'sampleId' },
+      });
+      const mockRes = httpMocks.createResponse();
+
+      const mockService = {
+        getById: jest.fn().mockReturnValueOnce({
+          sample: 'response',
+        }),
+      };
+
+      const controller = new CrudController({});
+      controller.service = mockService;
```

Comment:
might want to do this in a `beforeEach`.  All the methods in `mockService` can be defined in it and later in test we can just override the mocks.